mirrorlist/mirrorlist_server.py

```diff
@@ -161,7 +162,7 @@ def metalink(cache, directory, file, hosts_and_urls):
     preference = 100
     try:
         fdc = file_details_cache[directory]
-        detailslist = fdc[file]
+        detailslist = sorted(fdc[file], key=itemgetter('timestamp'), reverse=True)
```

Comment:
Good point. Although this list is usually not longer than 3 or 4 entries it would need to be sorted on every metalink request. Sorting it during pickle creation is definitely the right thing to do. I will submit a fixed PR.